gm8emulator/src/asset/sprite.rs

```diff
@@ -268,3 +268,25 @@ pub fn scale(input: &mut RgbaImage, width: u32, height: u32) {
         *input = RgbaImage::from_vec(width, height, output_vec).unwrap();
     }
 }
+
+impl Sprite {
+    fn get_image_index(&self, image_index: Real) -> Option<usize> {
+        (image_index.floor().into_inner() as isize).checked_rem_euclid(self.frames.len() as isize).map(|x| {x as usize})
```

Comment:
I trust you tested this doesn't do the funky rounding and actually floors the integer!